#! /bin/sh /usr/share/dpatch/dpatch-run
## 70_fix_milter_permissions.dpatch by Alex Prinsier <debian@aphexer.mailhaven.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: makes the milter tempdir group writeable so that av scanner can access it

@DPATCH@
diff -urNad pkg-amavisd-new~/helper-progs/amavis-milter.c pkg-amavisd-new/helper-progs/amavis-milter.c
--- pkg-amavisd-new~/helper-progs/amavis-milter.c	2007-10-07 09:13:35.000000000 +0200
+++ pkg-amavisd-new/helper-progs/amavis-milter.c	2007-10-07 09:18:29.000000000 +0200
@@ -210,6 +210,8 @@
 
 #ifdef HAVE_MKDTEMP
     stt = mkdtemp(s);
+    /* make the tempdir groupwriteable */
+    chmod(stt,S_IRWXU|S_IRGRP|S_IXGRP);
     if (stt == NULL)
 	amavis_syslog(DBG_FATAL, "(amavis_mkdtemp) mkdtemp %s failed: %s",
 		      s, strerror(errno));
